api/src/test/java/org/apache/iceberg/expressions/TestPredicateBinding.java

```diff
@@ -160,87 +158,81 @@ public void testLongToIntegerConversion() {
-        "Greater than below min should be alwaysTrue", Expressions.alwaysTrue(), gt.bind(struct));
+    assertThat(gt.bind(struct))
+        .as("Greater than below min should be alwaysTrue")
+        .isEqualTo(Expressions.alwaysTrue());
 
     UnboundPredicate<Long> gteq =
         new UnboundPredicate<>(GT_EQ, ref("i"), (long) Integer.MIN_VALUE - 1L);
-    Assert.assertEquals(
-        "Greater than or equal below min should be alwaysTrue",
-        Expressions.alwaysTrue(),
-        gteq.bind(struct));
+    assertThat(gteq.bind(struct))
+        .as("Greater than or equal below min should be alwaysTrue")
+        .isEqualTo(Expressions.alwaysTrue());
 
     UnboundPredicate<Long> gtMax =
         new UnboundPredicate<>(GT, ref("i"), (long) Integer.MAX_VALUE + 1L);
-    Assert.assertEquals(
-        "Greater than above max should be alwaysFalse",
-        Expressions.alwaysFalse(),
-        gtMax.bind(struct));
+    assertThat(gtMax.bind(struct))
+        .as("Greater than above max should be alwaysFalse")
+        .isEqualTo(Expressions.alwaysFalse());
 
     UnboundPredicate<Long> gteqMax =
         new UnboundPredicate<>(GT_EQ, ref("i"), (long) Integer.MAX_VALUE + 1L);
-    Assert.assertEquals(
-        "Greater than or equal above max should be alwaysFalse",
-        Expressions.alwaysFalse(),
-        gteqMax.bind(struct));
+    assertThat(gteqMax.bind(struct))
+        .as("Greater than or equal above max should be alwaysFalse")
+        .isEqualTo(Expressions.alwaysFalse());
 
     UnboundPredicate<Long> ltMin =
         new UnboundPredicate<>(LT, ref("i"), (long) Integer.MIN_VALUE - 1L);
-    Assert.assertEquals(
-        "Less than below min should be alwaysFalse", Expressions.alwaysFalse(), ltMin.bind(struct));
+    assertThat(ltMin.bind(struct))
+        .as("Less than below min should be alwaysFalse")
+        .isEqualTo(Expressions.alwaysFalse());
 
     UnboundPredicate<Long> lteqMin =
         new UnboundPredicate<>(LT_EQ, ref("i"), (long) Integer.MIN_VALUE - 1L);
-    Assert.assertEquals(
-        "Less than or equal below min should be alwaysFalse",
-        Expressions.alwaysFalse(),
-        lteqMin.bind(struct));
+    assertThat(lteqMin.bind(struct))
+        .as("Less than or equal below min should be alwaysFalse")
+        .isEqualTo(Expressions.alwaysFalse());
 
     Expression ltExpr =
         new UnboundPredicate<>(LT, ref("i"), (long) Integer.MAX_VALUE).bind(struct, true);
     BoundPredicate<Integer> ltMax = assertAndUnwrap(ltExpr);
-    Assert.assertTrue("Should be a literal predicate", ltMax.isLiteralPredicate());
-    Assert.assertEquals(
-        "Should translate bound to Integer",
-        (Integer) Integer.MAX_VALUE,
-        ltMax.asLiteralPredicate().literal().value());
+    assertThat(ltMax.isLiteralPredicate()).isTrue();
+    assertThat(ltMax.asLiteralPredicate().literal().value())
+        .as("Should translate bound to Integer")
+        .isEqualTo((Integer) Integer.MAX_VALUE);
```

Comment:
there are still type casts left like this one which we don't really need to do

Reply:
sry abt missing this. Removed this typecast

Reply:
sry for this back and forth. found the places where redundant typecasts were there.